providers/base/tests/test_pipewire_utils.py

```diff
@@ -0,0 +1,634 @@
+                     "object.serial": 29\
+                   },\
+                   "params": {\
+                       "Route": [{\
+                           "name": "hdmi_demo_output",\
+                           "available": "yes",\
+                           "description": "hdmi demo output"\
+                       }]\
+                   }\
+                 }\
+                }]'
+
+    @patch("subprocess.check_output")
+    def test_wrong_gst_pipeline_device(self, mock_checkout):
+        pt = PipewireTest()
+
+        mock_checkout.return_value = self.device
+        self.assertEqual(PipewireTestError.NO_SPECIFIC_DEVICE,
+                         pt.gst_pipeline("pipe", 10, "qoo"))
+
+    @patch("subprocess.check_output")
+    def test_gst_pipeline(self, mock_checkout):
+        pt = PipewireTest()
+
+        mock_checkout.return_value = self.device
+        self.assertEqual(PipewireTestError.NO_ERROR,
+                         pt.gst_pipeline("audiotestsrc wave=sine freq=512"
+                                         " ! audioconvert ! audioresample"
+                                         " ! autoaudiosink", 2, "hdmi"))
+
+
+class MonitorActivePortTests(unittest.TestCase):
+
+    # Correct Device
+    before_device = '[{\
+                 "id": 29,\
+                 "type": "PipeWire:Interface:Device",\
+                 "version": 3,\
+                 "permissions": [ "r", "w", "x", "m" ],\
+                 "info": {\
+                   "max-input-ports": 0,\
+                   "max-output-ports": 0,\
+                   "change-mask": [ "input-ports",\
+                                    "output-ports",\
+                                    "state", "props",\
+                                    "params" ],\
+                   "n-input-ports": 0,\
+                   "n-output-ports": 0,\
+                   "state": "suspended",\
+                   "error": null,\
+                   "props": {\
+                     "media.class": "Audio/Device",\
+                     "factory.name": "support.node.driver",\
+                     "node.name": "Freewheel-Driver",\
+                     "node.description": "unit test",\
+                     "priority.driver": 19000,\
+                     "node.group": "pipewire.freewheel",\
+                     "node.freewheel": true,\
+                     "factory.id": 10,\
+                     "clock.quantum-limit": 8192,\
+                     "node.driver": true,\
+                     "object.id": 29,\
+                     "object.serial": 29\
+                   },\
+                   "params": {\
+                       "Route": [{\
+                           "name": "hdmi_demo_output",\
+                           "available": "yes",\
+                           "direction": "Output",\
+                           "description": "hdmi demo output"\
+                       }]\
+                   }\
+                 }\
+                }]'
```

Comment:
please use multiline strings

Reply:
Change this to suggestion according to code review criteria, feel free to close it.